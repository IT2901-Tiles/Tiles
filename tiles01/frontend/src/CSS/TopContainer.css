/* MÅ FIKSES NÅR JEG HAR TID */ 
.TopContainer{
  display: block;
  padding-top: 1%;
  padding-bottom: 10%;
  padding-left: 1%;
  position: relative;
}

.Logo{
  background-color : #E53D2FEE;
  color: #F3EAC2;
  border-radius: 15px;
  font-family: 'Roboto';
  text-shadow: 0.25vw 0.25vw #252525ab;
  border: 0.25px solid #72140C;
  box-shadow: 1px 1px 3px #000000AA;
  min-height: 20px;
}

.Logo h1{
  padding: 0;
  position: center;
  margin: 2.5vh 0;
}


.TopContainer .Logo {
    width: 20vw;
    float: left;
    font-size: 1.5vw;
    margin-left: 2vw;
    min-height: 30px;
  }
  
.TopContainer .infoBox{
    background-color: rgba(108, 143, 143, 0.589);
    font-family: Helvetica, sans-serif;
    font-size: 1.5vw;
    color: #F3EAC2;
    border-radius: 15px;
    text-shadow: 0.20vw 0.20vw #252525ab;
    width: 60vw;
    position: absolute;
    right: 15vw;
    padding: 1.8vh 0;
    min-height: 30px;
  }

.TopContainer .infoBox p {
  padding: 0 10vw;
}

  /* Avoid the info-box from overlapping the .Logo */
@media only screen and (max-height: 800px) {
  .TopContainer .infoBox{
    margin-right: -4.5vw;
  } 
}