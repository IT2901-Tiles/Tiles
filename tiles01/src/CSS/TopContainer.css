/* MÅ FIKSES NÅR JEG HAR TID */ 
.TopContainer{
  display: block;
  padding-top: 1%;
  padding-bottom: 10%;
  padding-left: 1%;
}

.Logo{
  background-color : #E53D2FEE;
  color: #F3EAC2;
  border-radius: 2.5vw;
  font-family: 'Roboto';
  text-shadow: 2px 2px #252525ab;
  border: 0.25px solid #72140C;
  box-shadow: 1px 1px 3px #000000AA;
}

.Logo h1{
  padding: 0;
  position: center;
  margin: 3vh 0;
}


.TopContainer .Logo {
    width: 19rem;
    position: fixed;
    font-size: x-large;
  }
  
.TopContainer .infoBox{
    background-color: rgba(108, 143, 143, 0.589);
    font-family: 'Comic Neue', cursive;
    color: #F3EAC2;
    border-radius: 20px;
    text-shadow: 2px 2px #252525ab;
    width: 60%;
    height: 3.7rem;
    position: fixed;
    left: 21rem;
    padding: 1.5% 0;
  }
  