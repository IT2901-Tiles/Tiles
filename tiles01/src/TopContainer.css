/* MÅ FIKSES NÅR JEG HAR TID */ 
.TopContainer{
  display: block;
  padding-top: 1%;
  padding-bottom: 10%;
  padding-left: 1%;
}


.TopContainer .Logo {
    width: 20%;
    position: fixed;
    font-size: x-large;
  }
  
.TopContainer .infoBox{
    background-color: rgba(108, 143, 143, 0.589);
    font-family: 'Comic Neue', cursive;
    color: #F3EAC2;
    padding: 1%;
    border-radius: 20px;
    text-shadow: 2px 2px #252525ab;
    width: 60%;
    position: fixed;
    left: 25%;
    padding: 1.5% 0 1.5%;
  }
  