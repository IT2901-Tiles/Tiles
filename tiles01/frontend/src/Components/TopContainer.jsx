import React from 'react';
import "../CSS/TopContainer.css";
import Logo from "./Logo";


function TopContainer(props) {
    
    return (
        <div className="TopContainer">
            <Logo/>
            <div className="infoBox">
            <p>{props.text}</p>
            </div>
        </div>

    );
}

export default TopContainer;