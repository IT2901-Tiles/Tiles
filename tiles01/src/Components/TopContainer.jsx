import React from 'react';
//import "../App.css";
import "../TopContainer.css";
//import { Link } from "react-router-dom";
import Logo from "./Logo";


function TopContainer(props) {

    /* SKAL SLETTES HVIS EXTERN CSS FUNKER
    const infoBoxStyle ={
        backgroundColor: "rgba(108, 143, 143, 0.589)",
        fontFamily: "'Comic Neue', cursive",
        color: "#F3EAC2",
        padding: "1%",
        borderRadius: "20px",
        textShadow: "2px 2px #252525ab",
        width: "70%",
        position: "fixed",
        left: "25%",
        bottom: "88%",
    } */

    
    return (
        <div className="TopContainer">
            <Logo/>
            <div className="infoBox">
            <p>{props.text}</p>
            </div>
        </div>

    );
}

export default TopContainer;