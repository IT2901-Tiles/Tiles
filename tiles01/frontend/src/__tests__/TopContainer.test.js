import {configure, shallow} from "enzyme"
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import {cleanup} from '@testing-library/react';
import TopContainer from '../Components/TopContainer';
import renderer from 'react-test-renderer';

afterEach(() => {
    cleanup();
});

configure({
    adapter: new Adapter()
});

describe('TopContainer with no props', () => {
    const wrapper = shallow(<TopContainer/>)
    it('renders correctly', () => {
        const tree = renderer.create(<TopContainer/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('contains a Logo element', () => {
        expect(wrapper.find('Logo')).toHaveLength(1);
    });
    it('contains a p element', () => {
        expect(wrapper.find('p')).toHaveLength(1);
        expect(wrapper.find('p').text()).toEqual("");
    });
});

describe('TopContainer with props', () => {
    const initialProps = {text: 'hello, this is a test'};

    const container = shallow(<TopContainer {...initialProps} />);

    it('should have correct props for p element', () => {
        expect(container.find('p').props()).toEqual({"children": "hello, this is a test"});
    })
})