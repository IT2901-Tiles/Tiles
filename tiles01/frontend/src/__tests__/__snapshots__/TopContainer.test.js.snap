// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopContainer with no props renders correctly 1`] = `
<div
  className="TopContainer"
>
  <div
    className="Logo"
    style={
      Object {
        "backgroundColor": "#E53D2FEE",
        "border": "0.25px solid #72140C",
        "borderRadius": "30px",
        "boxShadow": "1px 1px 3px #000000AA",
        "color": "#F3EAC2",
        "fontFamily": "Roboto",
        "textShadow": "2px 2px #252525ab",
      }
    }
  >
    <h1
      style={
        Object {
          "margin": "15px 0",
          "padding": "0",
          "position": "center",
        }
      }
    >
      TILES
    </h1>
  </div>
  <div
    className="infoBox"
  >
    <p />
  </div>
</div>
`;
